src/firecracker/src/main.rs

```diff
@@ -281,6 +294,31 @@ fn print_supported_snapshot_versions() {
     println!("{}\n", snapshot_versions_str);
 }
 
+// Print data format of provided snapshot state file.
+fn print_snapshot_data_format(snapshot_path: &String) {
+    let mut snapshot_reader =
+        File::open(snapshot_path).expect("Unable to open snapshot state file");
+    let format_version_map = Snapshot::format_version_map();
+    let data_format_version =
+        Snapshot::get_format_version(&mut snapshot_reader, &format_version_map)
+            .expect("Invalid data format version of snapshot file");
+
+    let firecracker_data_version = FC_VERSION_TO_SNAP_VERSION.iter().find_map(|(key, &val)| {
+        if val == data_format_version {
+            Some(key)
+        } else {
+            None
+        }
+    });
+    match firecracker_data_version {
+        Some(val) => println!("v{}", val),
+        None => {
+            error!("{}", InvalidVersion);
```

Comment:
Can we give more information in the error message ?

Reply:
Done